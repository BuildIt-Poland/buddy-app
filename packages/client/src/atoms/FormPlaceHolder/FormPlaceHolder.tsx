import React from 'react';
import makeStyles from '@material-ui/core/styles/makeStyles';
import Skeleton from '@material-ui/lab/Skeleton';

const useStyles = makeStyles(theme => ({
  description: {
    height: '3rem',
    width: '20%',
  },
  input: {
    height: '6rem',
    margin: theme.spacing(0, 0, 1),
  },
  submit: {
    height: '7rem',
    margin: theme.spacing(3, 0),
    borderRadius: '25px',
  },
}));

const FormPlaceHolder: React.FC = () => {
  const classes = useStyles();

  const PlaceHolderTaskDescription = () => (
    <>
      <Skeleton variant={'text'} className={classes.description} />
      <Skeleton variant={'text'} className={classes.input} />
      <Skeleton variant={'text'} className={classes.description} />
      <Skeleton variant={'text'} className={classes.input} />
      <Skeleton variant={'text'} className={classes.description} />
      <Skeleton variant={'text'} className={classes.input} />
      <Skeleton variant={'text'} className={classes.submit} />
    </>
  );

  return <PlaceHolderTaskDescription />;
};

export default FormPlaceHolder;
