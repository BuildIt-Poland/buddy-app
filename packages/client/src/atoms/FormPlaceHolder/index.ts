import FormPlaceHolder from './FormPlaceHolder';

export default FormPlaceHolder;
