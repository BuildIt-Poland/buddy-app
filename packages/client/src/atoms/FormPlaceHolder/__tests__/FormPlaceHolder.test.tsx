import React from 'react';
import { create } from 'react-test-renderer';
import FormPlaceHolder from '../FormPlaceHolder';

jest.mock('@material-ui/lab/Skeleton', () => 'Skeleton');

describe('Component - FormPlaceHolder', () => {
  test('renders correctly', () => {
    const component = create(<FormPlaceHolder />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
