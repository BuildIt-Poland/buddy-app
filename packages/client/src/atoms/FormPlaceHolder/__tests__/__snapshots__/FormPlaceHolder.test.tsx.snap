// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - FormPlaceHolder renders correctly 1`] = `
Array [
  <Skeleton
    className="makeStyles-description-1"
    variant="text"
  />,
  <Skeleton
    className="makeStyles-input-2"
    variant="text"
  />,
  <Skeleton
    className="makeStyles-description-1"
    variant="text"
  />,
  <Skeleton
    className="makeStyles-input-2"
    variant="text"
  />,
  <Skeleton
    className="makeStyles-description-1"
    variant="text"
  />,
  <Skeleton
    className="makeStyles-input-2"
    variant="text"
  />,
  <Skeleton
    className="makeStyles-submit-3"
    variant="text"
  />,
]
`;
